entries/QUnit.module.xml

```diff
@@ -174,5 +177,46 @@ QUnit.module( "grouped tests argument hooks", function( hooks ) {
 } );
 ]]></code>
 	</example>
+	<example height="250">
+		<desc>An example of handling an asynchronous <code>then</code>able Promise result in hooks. This example uses an <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise">ES6 Promise</a> interface that is fulfilled after connecting to or disconnecting from database.</desc>
+<code><![CDATA[
+QUnit.module( "module A", {
+	before: function() {
+		var env = this;
+
+		// Set up connection to db
+		var promise = new Promise(function(resolve, reject) {
```

Comment:
Since this is just an example, can we shorten up this code snippet a bit? Maybe something like:

```js
before() {
  return new Promise(function(resolve, reject) {
    DB.connect(function(err) {
      if (err) {
        reject(err);
      } else {
        resolve();
      }
  });
}